examples/07_nvs_read_write/main/nvs_read_write.c

```diff
@@ -0,0 +1,62 @@
+/* Non-Volatile Storage (NVS) Read and Write Example
+
+   This example code is in the Public Domain (or CC0 licensed, at your option.)
+
+   Unless required by applicable law or agreed to in writing, this
+   software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+   CONDITIONS OF ANY KIND, either express or implied.
+*/
+#include <stdio.h>
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_system.h"
+#include "nvs_flash.h"
+#include "nvs.h"
+
+void app_main()
+{
+    nvs_flash_init();
+    system_init();
+
+    nvs_handle handle_to_settings;
+    esp_err_t err;
+    int32_t restart_counter = 0;
+
+    // Open the NVS
+    printf("Opening Non-Volatile Storage (NVS) ... ");
+    err = nvs_open("settings", NVS_READWRITE, &handle_to_settings);
+    printf((err != ESP_OK) ? "Failed!\n" : "OK\n");
```

Comment:
I did it on purpose to make the example simpler by not adding an extra `if` condition. My assumption was that issue with uninitialized `handle_to_settings` will be captured and displayed as an error in code below. Granted, I did not test this scenario.